codersdk/users.go

```diff
@@ -221,6 +221,27 @@ type OIDCAuthMethod struct {
 	IconURL    string `json:"iconUrl"`
 }
 
+type UserParameter struct {
+	Name  string `json:"name"`
+	Value string `json:"value"`
+}
+
+// UserAutofillParameters returns all recently used parameters for the given user.
+func (c *Client) UserAutofillParameters(ctx context.Context, user string, templateID string) ([]UserParameter, error) {
```

Comment:
`templateID` should be a `uuid.UUID` so the type checker can catch bugs with putting the wrong parameters.

`user` can't be a `uuid.UUID` since we also accept usernames and `"me"`, but it would be cool to define a type that encapsulates these options. I've seen at least one CLI bug where we naively accepted `username/workspacename` from CLI args and shoved it into a path parameter for `{workspace}` and then got strange results.